replica/database.cc

```diff
@@ -770,7 +770,7 @@ future<> database::modify_keyspace_on_all_shards(sharded<database>& sharded_db,
 future<> database::update_keyspace(const keyspace_metadata& tmp_ksm) {
     auto& ks = find_keyspace(tmp_ksm.name());
     auto new_ksm = ::make_lw_shared<keyspace_metadata>(tmp_ksm.name(), tmp_ksm.strategy_name(), tmp_ksm.strategy_options(), tmp_ksm.durable_writes(),
-                    boost::copy_range<std::vector<schema_ptr>>(ks.metadata()->cf_meta_data() | boost::adaptors::map_values), std::move(ks.metadata()->user_types()));
+                    boost::copy_range<std::vector<schema_ptr>>(ks.metadata()->cf_meta_data() | boost::adaptors::map_values), std::move(ks.metadata()->user_types()), tmp_ksm.get_storage_options());
```

Comment:
The real bug is that  keyspace_metadata has optional parameters.

Suggestions:
1. consolidate the constructors into one with non-default options
2. add a copy constructor

Reply:
On it, but some preliminary cleanup is waiting for its reviewers:
* #16447 
* #16449